dask/dataframe/tests/test_io.py

```diff
@@ -1107,6 +1107,24 @@ def test_read_hdf():
         assert (sorted(dd.read_hdf(fn, '/data').dask) ==
                 sorted(dd.read_hdf(fn, '/data').dask))
 
+def test_read_hdf_multiple():
+    pytest.importorskip('tables')
+    df = pd.DataFrame({'x': ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p'],
+                       'y': [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16]},
+                            index=[1., 2., 3., 4., 5., 6., 7., 8., 9., 10., 11., 12., 13., 14., 15., 16.])
+    a = dd.from_pandas(df, 16)
+
+    with tmpfile('h5') as fn:
+        a.to_hdf(fn, '/data*')
+        with pytest.raises(ValueError) as e:
+            r = dd.read_hdf(fn, '/data*', assume_partitioned=True)
+        assert 'entirety' in str(e)
```

Comment:
This is a bit ugly, but the current default value for `chunksize` is 1000000 or something.
While coding I made it a requirement to explicitly state `chunksize = None`.
It might needn't be a requirement now, and I'm ok with removing it as such.
The error is triggered [here](https://github.com/dask/dask/pull/1407/files#diff-2a4c3ed91854a1a70080d69581e762b3R682).

As i said this is all very WIP-iy and i tried making the interface very intolerant.

Do you think requiring `chunksize=None` is too much?